### PR TITLE
Give markers unique id's per graph

### LINKED ISSUE #1871: Theming of lineColor using directives spills over into subsequent diagrams
When setting the line color in a page this affects the styling of the arrowheads, the markers in svg. These ids are not properly contained within each svg. This results in the subsequent diagrams linking to the arrowhead definitions highest up in the page.

This can easily be reproduced by the example below:

image

```html
<html>
  <head>
  </head>
  <body>
  <h1>Example</h1>
  <div class="mermaid">
%%{init:{"theme":"base", "themeVariables": {"lineColor":"red"}}}%%
flowchart LR
subgraph red
A --> B
end
  </div>
  <div class="mermaid">
%%{init:{"theme":"base", "themeVariables": {"lineColor":"blue"}}}%%
flowchart LR
subgraph black
A --> B
end
  </div>
  <script src="//cdn.jsdelivr.net/npm/mermaid@8.9.0/dist/mermaid.min.js"></script>
  <!-- <script src="http://localhost:9000/mermaid.js"></script> -->
  <script>
      mermaid.initialize({
      });
      function callback(){alert('It worked');}
    </script>
</body>
</html>
```

### PR DESCRIPTION
**Pull request by chadfawcett**
## :bookmark_tabs: Summary

This PR makes markers have unique IDs based on the graph they are within. 

Resolves #1871
Supersedes #3546 - This PR seemed abandoned 

## :straight_ruler: Design Decisions

While #3546 seemed to resolve #1871, it appeared to be abandoned and also had a lot of syntax changes. I decided to keep the fix simple so hopefully it's easier to review and quicker to merge.

This implementation gives markers unique IDs based on the graph they are within. By using the ID, rather than some random string, we are able to reference the markers based on graph container ID when re-positioning. 

### :clipboard: Tasks

Make sure you

- [x] :book: have read the [contribution guidelines](https://github.com/mermaid-js/mermaid/blob/develop/CONTRIBUTING.md)
- [x] :computer: have added necessary unit/e2e tests.
- [ ] ~:notebook: have added documentation. Make sure [`MERMAID_RELEASE_VERSION`](https://github.com/mermaid-js/mermaid/blob/develop/packages/mermaid/src/docs/community/development.md#3-update-documentation) is used for all new features.~
- [x] :bookmark: targeted `develop` branch

### CODE REVIEW

**Review by sidharthv96 — APPROVED** [review]
@chadfawcett the changes looks great!

Can you please add an integration test to verify that the issue is solved?

The content from https://github.com/mermaid-js/mermaid/issues/1871 can be in the html file. 
You can refer https://github.com/mermaid-js/mermaid/blob/bf18f25c820fb9264fe70ce5113a85f676cfd306/cypress/integration/other/ghsa.spec.js on how it should be done.

**Comment by chadfawcett** [comment]
@sidharthv96 I didn't see any documentation around contributing to the tests, so I did my best guess at what you might want. It'd be nice to see some documentation about what you expect in the [CONTRIBUTING.md ](https://github.com/mermaid-js/mermaid/blob/00d06c7282a701849793680c1e97da1cfdfcce62/CONTRIBUTING.md)doc with best practices around naming and file location etc.

Here's a screenshot of what I see in Cypress. Let me know if you need me to make changes to what I've done.

Screenshot 2023-09-11 at 11 48 12 AM

File: cypress/platform/marker_unique_id.html [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      A --> B
+      end
+    </pre>
+    <pre class="mermaid">
+      %%{init:{"theme":"base", "themeVariables": {"lineColor":"blue"}}}%%
+      flowchart LR
+      subgraph black
+      A --> B
+      end
+    </pre>
```
Comment (sidharthv96) [inline cypress/platform/marker_unique_id.html:18]: Let's add the new yaml config option to the test. 

```suggestion
    <pre class="mermaid">
      %%{init:{"theme":"base", "themeVariables": {"lineColor":"red"}}}%%
      flowchart LR
      subgraph red
      A --> B
      end
    </pre>
    <pre class="mermaid">
      %%{init:{"theme":"base", "themeVariables": {"lineColor":"blue"}}}%%
      flowchart LR
      subgraph black
      A --> B
      end
    </pre>
    <pre class="mermaid">
      ---
      config:
        theme: base
        themeVariables:
          lineColor: yellow
      ---
      flowchart LR
      subgraph red
      A --> B
      end
    </pre>
    <pre class="mermaid">
      ---
      config:
        theme: base
        themeVariables:
          lineColor: green
      ---
      flowchart LR
      subgraph black
      A --> B
      end
    </pre>
```

**Review by sidharthv96 — APPROVED** [review]
The test is great. 
I've added 2 more cases.

**Comment by chadfawcett** [comment]
Thanks @sidharthv96 and @knsv for getting this merged in. Any idea when I could expect this fix to be released in a new version on npm?

**Comment by sidharthv96** [comment]
Our plan was to release today, but we discovered some bugs while testing. So that has blocked the release.

**Pull request merged**